src/frontend/js/modules/authentication.js

```diff
@@ -50,7 +50,8 @@ angular.module('esn.authentication', ['esn.http'])
       if (!expirationTime) return promise || requestNewWebToken();
 
       // check if we haven't reached the expiration time yet.
-      return expirationTime > Date.now() ? promise : requestNewWebToken();
+      // jwt exp claim is a NumericDate value that is defined as the number of seconds (not milliseconds) since Epoch.
+      return expirationTime > Date.now() / 1000 ? promise : requestNewWebToken();
```

Comment:
sure but how we can work if its not behind OIDC ?

Reply:
this has nothing to do with OIDC i believe, client ( ie calendar SPA ) will request a `jwt` token from ESN to use it when making requests to services like  `Sabre` and `jmap`.

Reply:
Ok great. Are we saying this have Never worked due to this error

Reply:
well it was working, but the error was that it didn't properly caching the jwt tokens, it was requesting new tokens too often instead of using the cache.